src/probly/layers/flax.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+import jax
+from flax import nnx
+from flax.nnx import rnglib   
+
+
+class DropConnectDense(nnx.Module):
+ 
+ def __init__(self, base_layer: nnx.Linear, rngs: rnglib.Rngs, p: float = 0.25):
+    """Custom Linear layer with DropConnect applied to weights during training.
+
+    Attributes:
+        rngs:           flax.nnx.Rngs, The JAX RNG state abstraction.
+        in_features:    int, The number of input features for the layer.
+        out_features:   int, The number of output features for the layer.
+        p:              float, The probability of dropping an individual weight connection.
+        weight:         flax.nnx.Param, The weight matrix of the layer with shape `(out_features, in_features)`.
+        bias:           flax.nnx.Param | None, The bias vector of the layer, or None if it is not used.
+
+    """
+    self.rngs = rngs
+    self.in_features = base_layer.kernel.shape[0]
+    self.out_features = base_layer.kernel.shape[1]
+    self.p = p
+    self.weight = nnx.Param(base_layer.kernel.T)
+
+    if base_layer.use_bias:
+        self.bias = nnx.Param(base_layer.bias)
+    else:
+        self.bias = None
+    
+ def __call__(self, x: jax.Array, *, deterministic: bool = False) -> jax.Array:
+     if not deterministic:
+        key = self.rngs.dropout()
+        mask = jax.random.uniform(key, shape=self.weight.value.shape) > self.p
+        weight = self.weight.value * mask  
+     else:
+        weight = self.weight.value * (1.0 - self.p)
+     y = x @ weight.T
+     if self.bias is not None:
```

Comment:
`__call__` should optionally also accept `rngs`. This allows the used to provide a PRNG either at construction or at call time.
Providing the PRNG at call time is the preferred way of doing this in flax as it allows for more explicit control over stochasticity (see [Randomness in flax](https://flax.readthedocs.io/en/latest/guides/randomness.html)).

Take a look at the implementation of `Dropout` in flax. The implementation of `DropConnectDense` should roughly follow it.

```suggestion
    def __call__(
        self,
        x: jax.Array,
        *,
        deterministic: bool = False,
        rngs: rnglib.Rngs | rnglib.RngStream | jax.Array | None = None,
    ) -> jax.Array:
```